cli/commands/list/list.go

```diff
@@ -209,7 +233,19 @@ func discoveredToListed(components component.Components, opts *Options) (ListedC
 		listedCfg.Dependencies = make([]*ListedComponent, len(c.Dependencies()))
 
 		for i, dep := range c.Dependencies() {
-			relDepPath, err := filepath.Rel(opts.WorkingDir, dep.Path())
+			// Ensure dependency path is absolute for reliable relativity calculation
+			depPath := dep.Path()
```

Comment:
Wait, why is the path ever relative here? I feel like we should fix the point of ingestion to make it so that components reliably store their paths as absolute paths.